### ISSUE TITLE
arm linux getdents is incorrect

### ISSUE THREAD

**Issue by lunixbochs** [lunixbochs/usercorn#164]

**Comment by lunixbochs** [comment]
might be wrong, a quick trace showed getdents wasn't even called, just an `lstat(".")`

**Comment by petevine** [comment]
You're right of course - the displayed dot is coloured red which means there're some garbage terminal codes?

I've just noticed the working (x86_64) `ls -la` doesn't display `.` and `..` elements.

**Comment by lunixbochs** [comment]
Oh, I need to manually add . and .. - opened #170

**Comment by lunixbochs** [comment]
This might be the problem :) https://github.com/lunixbochs/usercorn/blob/master/go/kernel/posix/stat.go#L16
